cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -283,3 +248,50 @@ def _serialize_entrance_exam(entrance_exam_module):
     return json.dumps({
         'locator': unicode(entrance_exam_module.location)
     })
+
+
+def add_entrance_exam_milestone(course_id, x_block):
```

Comment:
Yes, sure i will add some tests as i mentioned. keeping it in WIP because i have few questions and concerns to clear first.